From: Robert Mader <robert.mader@posteo.de>
Date: Sun, 2 Jan 2022 20:55:15 +0100
Subject: display-config: Do not invert order when constructing modes

Mutter sends modes in descending order of preference. By reverting
the order via `g_list_prepend`, we get unintended side effects
such as choosing the least preferred refresh rate by default (if
the selected mode is not marked as preferred).

Instead of adding complex logic in several places, make sure that
the assumption of descending preference is kept by simply not
inverting the order.

(cherry picked from commit 6a4652bd35e23a4ae4ec3078be380ed09858e0ee)

Bug: https://gitlab.gnome.org/GNOME/gnome-control-center/-/issues/1562
Origin: upstream, 41.3, commit:293e191e399123c91ef5d7b5c796ea0f42b8bd91
---
 panels/display/cc-display-config-dbus.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/panels/display/cc-display-config-dbus.c b/panels/display/cc-display-config-dbus.c
index ae9eb01..bc20b69 100644
--- a/panels/display/cc-display-config-dbus.c
+++ b/panels/display/cc-display-config-dbus.c
@@ -795,6 +795,8 @@ construct_modes (CcDisplayMonitorDBus *self,
       if (mode->flags & MODE_CURRENT)
         self->current_mode = CC_DISPLAY_MODE (mode);
     }
+
+  self->modes = g_list_reverse (self->modes);
 }
 
 static CcDisplayMonitorDBus *
